### ISSUE TITLE
linter: vue/no-required-prop-with-default false positive

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#14072]
> This one is also false positive:
> 
> ```ts
> interface Props {
>   readonly phoneNumber: string;
> }
> 
> const { phoneNumber } = defineProps<Props>();
> ```
> 
> ```shell
>   × eslint-plugin-vue(no-required-prop-with-default): Prop "phoneNumber" should be optional.
>     ╭─[component.vue:19:5]
>  18 │   interface Props {
>  19 │     readonly phoneNumber: string;
>     ·     ─────────────────────────────
>  20 │   }
>     ╰────
>   help: Remove the `required: true` option, or drop the `required` key entirely to make this prop optional.
> ``` 

 _Originally posted by @Perdolique in [#14056](https://github.com/oxc-project/oxc/issues/14056#issuecomment-3327184352)_

**Comment by huangtiandi1999** [comment]
I will take a look！

**Comment by huangtiandi1999** [comment]
@camc314 This PR (#14066) also addresses this issue.

**Comment by camc314** [comment]
ah perfect, thanks
